### DISCUSSION TITLE
The "line shift" feature in Lem ... what is it?

### DISCUSSION THREAD

**Discussion by aykaramba** [lem-project/lem#907] [General]
As I am using Lem I am learning more about it.  One interesting function is the feature that shifts a single line of code to the left in a piece of code that is too long for the display area so that you can see the entire line of code and edit it in place without shifting your entire document view to the right.

What is this feature called?  Is it possible to turn the feature off temporarily?

I am not sure what to think of the feature just yet.  I understand why it is useful, however, when the feature gets triggered and I am QUICKLY scrolling through lines of code or clicking around on different lines of code, it becomes visually jarring to have lines of code jump around horizontally and makes it really difficult to traverse code.

If there is a way to turn that off / on that would help me test the feature with my workflow so that I can better understand it's pros and cons.

Thanks.

**Comment by cxxxr** [comment]
Thank you for the feedback.

I do not know if there is a special way to call it.
This feature mimics older editors (jed, micro emacs, etc.).

Although this behavior has the advantage of being easier to implement and requiring fewer resources than is reasonable, I believe it should be changed to a general horizontal scrolling behavior.

**Comment by aykaramba** [comment]
Thank you kindly for reading my post and considering it.  Please note that I have not really made up my mind and I don't want anyone to lose the feature if they want it either.  It may be the right way to do things.  

Right now I am just a noob and I like to jump around and scrolls through code quickly and in that specific use case it is just a little bit jarring.

Thank you for the amazing editor!
